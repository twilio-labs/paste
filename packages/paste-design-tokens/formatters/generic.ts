import type {ImmutableStyleMap} from 'theo';
import type {DesignToken, GenericDesignToken} from '../types';
import {getTokenCategories} from '../utils/getTokenCategories';
import {formatGroupTokensWithTemplate} from '../utils/formatGroupTokensWithTemplate';
import {rgbToHex} from '../utils/rgbToHex';
import {remToPx} from '../utils/remToPx';

export const categoryTemplate = (categoryName: string, props: DesignToken[]): string => {
  const tokenPairs = props.map(({category, ...rest}) => rest);

  return `
    "${categoryName}": ${JSON.stringify(tokenPairs)},
  `;
};

const getTokenAltValue = ({category, value}: {category: string; value: string}): string | null => {
  switch (category) {
    case 'background-color':
    case 'border-color':
    case 'text-color':
      return typeof value === 'string' ? rgbToHex(value) : null;
    case 'font-size':
    case 'line-height':
    case 'sizing':
    case 'spacing':
      return remToPx(value, 'string') as string;
    default:
      return null;
  }
};

// Filter out deprecated tokens in exported list
// Add altValues in exported list
// Remove originalValue (shouldn't reveal how the sausage is made)
export const tweakTokens = (tokenCategory: DesignToken[]): GenericDesignToken[] => {
  const tokensWithoutDeprecated = tokenCategory.filter((token) => !token.deprecated);
  const tweakedTokenShape = tokensWithoutDeprecated.map(({originalValue, ...token}) => ({
    ...token,
    altValue: getTokenAltValue(token),
  }));

  return tweakedTokenShape;
};

export const genericCommonJsTokenFormat = (result: ImmutableStyleMap): string => {
  const categories = getTokenCategories(result);

  const groupedTokens = `"tokens":{${formatGroupTokensWithTemplate(
    result,
    categories,
    categoryTemplate,
    tweakTokens
  )}}`;

  return `module.exports = {\n${groupedTokens}\n}`;
};
