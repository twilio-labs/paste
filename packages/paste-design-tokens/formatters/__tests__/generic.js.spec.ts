import {resolve} from 'path';

import * as theo from 'theo';

import {genericCommonJsTokenFormat} from '../generic';

theo.registerFormat('generic.js', genericCommonJsTokenFormat);

describe('generic JS formatter', () => {
  it('should return commonjs formatted tokens', async () => {
    return theo
      .convert({
        transform: {
          type: 'web',
          file: resolve(__dirname, '../__fixtures__/index.yml'),
        },
        format: {
          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
          // @ts-ignore Theo isn't typed for custom format types
          type: 'generic.js',
        },
      })
      .then((genericJS: string) => {
        return expect(genericJS).toMatchSnapshot();
      })
      .catch((error: string) => {
        // eslint-disable-next-line no-console
        console.log(`Something  went wrong: ${error}`);
        throw new Error('[es6Formatter test]: should return es6 formatted tokens');
      });
  });
});
