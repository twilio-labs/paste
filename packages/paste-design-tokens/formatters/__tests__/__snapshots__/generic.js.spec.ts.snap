// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generic JS formatter should return commonjs formatted tokens 1`] = `
"module.exports = {
\\"tokens\\":{
    \\"background-colors\\": [{\\"type\\":\\"color\\",\\"value\\":\\"rgb(255, 255, 255)\\",\\"comment\\":\\"Some background\\",\\"originalValue\\":\\"{!white}\\",\\"name\\":\\"color_background_body\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(242, 47, 70)\\",\\"comment\\":\\"Primary brand background, accessible with black text\\",\\"originalValue\\":\\"{!amaranth}\\",\\"name\\":\\"color_background_brand_highlight\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(235, 15, 41)\\",\\"comment\\":\\"Dark Primary brand background, accessible with black and white text\\",\\"originalValue\\":\\"{!red_ribbon}\\",\\"name\\":\\"color_background_brand_highlight_dark\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(194, 37, 55)\\",\\"comment\\":\\"Darker Primary brand background, accessible with black and white text\\",\\"originalValue\\":\\"{!palette_red_50}\\",\\"name\\":\\"color_background_brand_highlight_darker\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(13, 18, 43)\\",\\"comment\\":\\"Primary brand background, accessible with black text\\",\\"originalValue\\":\\"{!haiti}\\",\\"name\\":\\"color_background_brand_primary\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(246, 246, 246)\\",\\"comment\\":\\"Some background\\",\\"originalValue\\":\\"{!palette_gray_20}\\",\\"name\\":\\"color_background_default\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(236, 187, 193)\\",\\"comment\\":\\"Some background\\",\\"originalValue\\":\\"{!palette_red_20}\\",\\"name\\":\\"color_background_element_selection\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(206, 80, 95)\\",\\"comment\\":\\"Some background\\",\\"originalValue\\":\\"{!palette_red_40}\\",\\"name\\":\\"color_background_error\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(174, 33, 49)\\",\\"comment\\":\\"Some background\\",\\"originalValue\\":\\"{!palette_red_60}\\",\\"name\\":\\"color_background_error_dark\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(194, 37, 55)\\",\\"comment\\":\\"Some background\\",\\"originalValue\\":\\"{!palette_red_50}\\",\\"name\\":\\"color_background_error_light\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(230, 230, 230)\\",\\"comment\\":\\"Some background\\",\\"originalValue\\":\\"{!palette_gray_40}\\",\\"name\\":\\"color_background_row_active\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(214, 214, 214)\\",\\"comment\\":\\"Some background\\",\\"originalValue\\":\\"{!palette_gray_50}\\",\\"name\\":\\"color_background_row_focus\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(214, 214, 214)\\",\\"comment\\":\\"Some background\\",\\"originalValue\\":\\"{!palette_gray_50}\\",\\"name\\":\\"color_background_row_hover\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(230, 230, 230)\\",\\"comment\\":\\"Some background\\",\\"originalValue\\":\\"{!palette_gray_40}\\",\\"name\\":\\"color_background_row_selected\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(235, 235, 235)\\",\\"comment\\":\\"Some background\\",\\"originalValue\\":\\"{!palette_gray_30}\\",\\"name\\":\\"color_background_row_striped\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(100, 209, 138)\\",\\"comment\\":\\"Some background\\",\\"originalValue\\":\\"{!palette_green_50}\\",\\"name\\":\\"color_background_success\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(80, 167, 110)\\",\\"comment\\":\\"Some background\\",\\"originalValue\\":\\"{!palette_green_70}\\",\\"name\\":\\"color_background_success_dark\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(162, 227, 184)\\",\\"comment\\":\\"Some background\\",\\"originalValue\\":\\"{!palette_green_30}\\",\\"name\\":\\"color_background_success_light\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(251, 208, 144)\\",\\"comment\\":\\"Some background\\",\\"originalValue\\":\\"{!palette_orange_30}\\",\\"name\\":\\"color_background_warning\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(199, 141, 56)\\",\\"comment\\":\\"Some background\\",\\"originalValue\\":\\"{!palette_orange_70}\\",\\"name\\":\\"color_background_warning_dark\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(249, 177, 71)\\",\\"comment\\":\\"Some background\\",\\"originalValue\\":\\"{!palette_orange_50}\\",\\"name\\":\\"color_background_warning_light\\"}],
  

    \\"colors\\": [{\\"type\\":\\"color\\",\\"value\\":\\"rgb(242, 47, 70)\\",\\"comment\\":\\"Twilio brand red\\",\\"originalValue\\":\\"{!amaranth}\\",\\"name\\":\\"color_brand_highlight\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(235, 15, 41)\\",\\"comment\\":\\"Twilio accessible brand red\\",\\"originalValue\\":\\"{!red_ribbon}\\",\\"name\\":\\"color_brand_highlight_dark\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(13, 18, 43)\\",\\"comment\\":\\"Primary branding color\\",\\"originalValue\\":\\"{!haiti}\\",\\"name\\":\\"color_brand_primary\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(255, 255, 255)\\",\\"comment\\":\\"Gray Color 1\\",\\"originalValue\\":\\"{!palette_gray_10}\\",\\"name\\":\\"color_gray_1\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(246, 246, 246)\\",\\"comment\\":\\"Gray Color 2\\",\\"originalValue\\":\\"{!palette_gray_20}\\",\\"name\\":\\"color_gray_2\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(235, 235, 235)\\",\\"comment\\":\\"Gray Color 3\\",\\"originalValue\\":\\"{!palette_gray_30}\\",\\"name\\":\\"color_gray_3\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(230, 230, 230)\\",\\"comment\\":\\"Gray Color 4\\",\\"originalValue\\":\\"{!palette_gray_40}\\",\\"name\\":\\"color_gray_4\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(214, 214, 214)\\",\\"comment\\":\\"Gray Color 5\\",\\"originalValue\\":\\"{!palette_gray_50}\\",\\"name\\":\\"color_gray_5\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(204, 204, 204)\\",\\"comment\\":\\"Gray Color 6\\",\\"originalValue\\":\\"{!palette_gray_60}\\",\\"name\\":\\"color_gray_6\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(153, 153, 153)\\",\\"comment\\":\\"Gray Color 7\\",\\"originalValue\\":\\"{!palette_gray_70}\\",\\"name\\":\\"color_gray_7\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(102, 102, 102)\\",\\"comment\\":\\"Gray Color 8\\",\\"originalValue\\":\\"{!palette_gray_80}\\",\\"name\\":\\"color_gray_8\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(68, 68, 68)\\",\\"comment\\":\\"Gray Color 9\\",\\"originalValue\\":\\"{!palette_gray_90}\\",\\"name\\":\\"color_gray_9\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(34, 34, 34)\\",\\"comment\\":\\"Gray Color 10\\",\\"originalValue\\":\\"{!palette_gray_100}\\",\\"name\\":\\"color_gray_10\\"}],
  

    \\"font-sizes\\": [{\\"type\\":\\"font-size\\",\\"value\\":\\"0.6875rem\\",\\"comment\\":\\"Constant typography token for font size 0\\",\\"originalValue\\":\\"{!font_size_0}\\",\\"name\\":\\"font_size_0\\"},{\\"type\\":\\"font-size\\",\\"value\\":\\"0.75rem\\",\\"comment\\":\\"Constant typography token for font size 1\\",\\"originalValue\\":\\"{!font_size_1}\\",\\"name\\":\\"font_size_1\\"},{\\"type\\":\\"font-size\\",\\"value\\":\\"0.8125rem\\",\\"comment\\":\\"Constant typography token for font size 2\\",\\"originalValue\\":\\"{!font_size_2}\\",\\"name\\":\\"font_size_2\\"},{\\"type\\":\\"font-size\\",\\"value\\":\\"0.875rem\\",\\"comment\\":\\"Constant typography token for font size 3\\",\\"originalValue\\":\\"{!font_size_3}\\",\\"name\\":\\"font_size_3\\"},{\\"type\\":\\"font-size\\",\\"value\\":\\"0.9375rem\\",\\"comment\\":\\"Constant typography token for font size 4\\",\\"originalValue\\":\\"{!font_size_4}\\",\\"name\\":\\"font_size_4\\"},{\\"type\\":\\"font-size\\",\\"value\\":\\"1rem\\",\\"comment\\":\\"Constant typography token for font size 5\\",\\"originalValue\\":\\"{!font_size_5}\\",\\"name\\":\\"font_size_5\\"},{\\"type\\":\\"font-size\\",\\"value\\":\\"1.125rem\\",\\"comment\\":\\"Constant typography token for font size 6\\",\\"originalValue\\":\\"{!font_size_6}\\",\\"name\\":\\"font_size_6\\"},{\\"type\\":\\"font-size\\",\\"value\\":\\"1.25rem\\",\\"comment\\":\\"Constant typography token for font size 7\\",\\"originalValue\\":\\"{!font_size_7}\\",\\"name\\":\\"font_size_7\\"},{\\"type\\":\\"font-size\\",\\"value\\":\\"1.875rem\\",\\"comment\\":\\"Constant typography token for font size 8\\",\\"originalValue\\":\\"{!font_size_8}\\",\\"name\\":\\"font_size_8\\"},{\\"type\\":\\"font-size\\",\\"value\\":\\"2.25rem\\",\\"comment\\":\\"Constant typography token for font size 9\\",\\"originalValue\\":\\"{!font_size_9}\\",\\"name\\":\\"font_size_9\\"}],
  

    \\"spacings\\": [{\\"type\\":\\"size\\",\\"value\\":\\"0.25rem\\",\\"originalValue\\":\\"{!space_0}\\",\\"name\\":\\"space_0\\"},{\\"type\\":\\"size\\",\\"value\\":\\"0.5rem\\",\\"originalValue\\":\\"{!space_1}\\",\\"name\\":\\"space_1\\"},{\\"type\\":\\"size\\",\\"value\\":\\"0.75rem\\",\\"originalValue\\":\\"{!space_2}\\",\\"name\\":\\"space_2\\"},{\\"type\\":\\"size\\",\\"value\\":\\"1rem\\",\\"originalValue\\":\\"{!space_3}\\",\\"name\\":\\"space_3\\"},{\\"type\\":\\"size\\",\\"value\\":\\"1.25rem\\",\\"originalValue\\":\\"{!space_4}\\",\\"name\\":\\"space_4\\"},{\\"type\\":\\"size\\",\\"value\\":\\"1.5rem\\",\\"originalValue\\":\\"{!space_5}\\",\\"name\\":\\"space_5\\"},{\\"type\\":\\"size\\",\\"value\\":\\"1.75rem\\",\\"originalValue\\":\\"{!space_6}\\",\\"name\\":\\"space_6\\"},{\\"type\\":\\"size\\",\\"value\\":\\"2rem\\",\\"originalValue\\":\\"{!space_7}\\",\\"name\\":\\"space_7\\"},{\\"type\\":\\"size\\",\\"value\\":\\"2.25rem\\",\\"originalValue\\":\\"{!space_8}\\",\\"name\\":\\"space_8\\"},{\\"type\\":\\"size\\",\\"value\\":\\"2.5rem\\",\\"originalValue\\":\\"{!space_9}\\",\\"name\\":\\"space_9\\"},{\\"type\\":\\"size\\",\\"value\\":\\"2.75rem\\",\\"originalValue\\":\\"{!space_10}\\",\\"name\\":\\"space_10\\"},{\\"type\\":\\"size\\",\\"value\\":\\"3rem\\",\\"originalValue\\":\\"{!space_11}\\",\\"name\\":\\"space_11\\"},{\\"type\\":\\"size\\",\\"value\\":\\"3.25rem\\",\\"originalValue\\":\\"{!space_12}\\",\\"name\\":\\"space_12\\"},{\\"type\\":\\"size\\",\\"value\\":\\"3.5rem\\",\\"originalValue\\":\\"{!space_13}\\",\\"name\\":\\"space_13\\"},{\\"type\\":\\"size\\",\\"value\\":\\"3.75rem\\",\\"originalValue\\":\\"{!space_14}\\",\\"name\\":\\"space_14\\"},{\\"type\\":\\"size\\",\\"value\\":\\"4rem\\",\\"originalValue\\":\\"{!space_15}\\",\\"name\\":\\"space_15\\"}],
  

    \\"text-colors\\": [{\\"type\\":\\"color\\",\\"value\\":\\"rgb(242, 47, 70)\\",\\"comment\\":\\"Twilio brand red, accessible on large text only.\\",\\"originalValue\\":\\"{!amaranth}\\",\\"name\\":\\"color_text_brand_highlight\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(235, 15, 41)\\",\\"comment\\":\\"Accessible Twilio brand red.\\",\\"originalValue\\":\\"{!red_ribbon}\\",\\"name\\":\\"color_text_brand_highlight_dark\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(255, 255, 255)\\",\\"comment\\":\\"Text color used on any brand color\\",\\"originalValue\\":\\"{!white}\\",\\"name\\":\\"color_text_brand_primary\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(68, 68, 68)\\",\\"comment\\":\\"Body text color\\",\\"originalValue\\":\\"{!palette_gray_90}\\",\\"name\\":\\"color_text_default\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(194, 37, 55)\\",\\"comment\\":\\"Error text for inputs and error misc\\",\\"originalValue\\":\\"{!palette_red_50}\\",\\"name\\":\\"color_text_error\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(68, 68, 68)\\",\\"comment\\":\\"Default icon color.\\",\\"originalValue\\":\\"{!palette_gray_90}\\",\\"name\\":\\"color_text_icon_default\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(255, 255, 255)\\",\\"comment\\":\\"Inverse text color for dark backgrounds\\",\\"originalValue\\":\\"{!white}\\",\\"name\\":\\"color_text_inverse\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(235, 235, 235)\\",\\"comment\\":\\"Weak inverse text color for dark backgrounds\\",\\"originalValue\\":\\"{!palette_gray_30}\\",\\"name\\":\\"color_text_inverse_weak\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(68, 68, 68)\\",\\"comment\\":\\"Text color for field labels.\\",\\"originalValue\\":\\"{!palette_gray_90}\\",\\"name\\":\\"color_text_label\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(0, 112, 204)\\",\\"comment\\":\\"Link text\\",\\"originalValue\\":\\"{!palette_blue_50}\\",\\"name\\":\\"color_text_link\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(0, 89, 163)\\",\\"comment\\":\\"Active link text\\",\\"originalValue\\":\\"{!palette_blue_70}\\",\\"name\\":\\"color_text_link_active\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(194, 37, 55)\\",\\"comment\\":\\"Destructive link text\\",\\"originalValue\\":\\"{!palette_red_50}\\",\\"name\\":\\"color_text_link_destructive\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(155, 29, 44)\\",\\"comment\\":\\"Active destructive link text\\",\\"originalValue\\":\\"{!palette_red_70}\\",\\"name\\":\\"color_text_link_destructive_active\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(236, 187, 193)\\",\\"comment\\":\\"Disabled destructive link text\\",\\"originalValue\\":\\"{!palette_red_20}\\",\\"name\\":\\"color_text_link_destructive_disabled\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(174, 33, 49)\\",\\"comment\\":\\"Focus destructive link text\\",\\"originalValue\\":\\"{!palette_red_60}\\",\\"name\\":\\"color_text_link_destructive_focus\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(174, 33, 49)\\",\\"comment\\":\\"Hover destructive link text\\",\\"originalValue\\":\\"{!palette_red_60}\\",\\"name\\":\\"color_text_link_destructive_hover\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(102, 169, 224)\\",\\"comment\\":\\"Disabled link text\\",\\"originalValue\\":\\"{!palette_blue_30}\\",\\"name\\":\\"color_text_link_disabled\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(0, 97, 192)\\",\\"comment\\":\\"Focus link text\\",\\"originalValue\\":\\"{!palette_blue_60}\\",\\"name\\":\\"color_text_link_focus\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(0, 97, 192)\\",\\"comment\\":\\"Hover link text\\",\\"originalValue\\":\\"{!palette_blue_60}\\",\\"name\\":\\"color_text_link_hover\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(102, 102, 102)\\",\\"comment\\":\\"Input placeholder text.\\",\\"originalValue\\":\\"{!palette_gray_80}\\",\\"name\\":\\"color_text_placeholder\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(100, 209, 138)\\",\\"comment\\":\\"Text color for success text.\\",\\"originalValue\\":\\"{!palette_green_50}\\",\\"name\\":\\"color_text_success\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(249, 177, 71)\\",\\"comment\\":\\"Color for texts or icons that are related to warnings on a dark background.\\",\\"originalValue\\":\\"{!palette_orange_50}\\",\\"name\\":\\"color_text_warning\\"},{\\"type\\":\\"color\\",\\"value\\":\\"rgb(230, 230, 230)\\",\\"comment\\":\\"Weak body text for visual hierarchy.\\",\\"originalValue\\":\\"{!palette_gray_40}\\",\\"name\\":\\"color_text_weak\\"}],
  

    \\"z-indices\\": [{\\"type\\":\\"z-index\\",\\"value\\":\\"0\\",\\"originalValue\\":\\"{!z-index-0}\\",\\"name\\":\\"z-index-0\\"},{\\"type\\":\\"z-index\\",\\"value\\":\\"10\\",\\"originalValue\\":\\"{!z-index-10}\\",\\"name\\":\\"z-index-10\\"},{\\"type\\":\\"z-index\\",\\"value\\":\\"20\\",\\"originalValue\\":\\"{!z-index-20}\\",\\"name\\":\\"z-index-20\\"},{\\"type\\":\\"z-index\\",\\"value\\":\\"30\\",\\"originalValue\\":\\"{!z-index-30}\\",\\"name\\":\\"z-index-30\\"},{\\"type\\":\\"z-index\\",\\"value\\":\\"40\\",\\"originalValue\\":\\"{!z-index-40}\\",\\"name\\":\\"z-index-40\\"},{\\"type\\":\\"z-index\\",\\"value\\":\\"50\\",\\"originalValue\\":\\"{!z-index-50}\\",\\"name\\":\\"z-index-50\\"},{\\"type\\":\\"z-index\\",\\"value\\":\\"60\\",\\"originalValue\\":\\"{!z-index-60}\\",\\"name\\":\\"z-index-60\\"},{\\"type\\":\\"z-index\\",\\"value\\":\\"70\\",\\"originalValue\\":\\"{!z-index-70}\\",\\"name\\":\\"z-index-70\\"},{\\"type\\":\\"z-index\\",\\"value\\":\\"80\\",\\"originalValue\\":\\"{!z-index-80}\\",\\"name\\":\\"z-index-80\\"},{\\"type\\":\\"z-index\\",\\"value\\":\\"90\\",\\"originalValue\\":\\"{!z-index-90}\\",\\"name\\":\\"z-index-90\\"}],
  }
}"
`;
